### ISSUE TITLE
no-unused-expressions doesn't recognize void

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#431]
The `no-unused-expressions` rule incorrectly flags this pattern:

``` js
void new RegExp();
```

The presence of `void` is an indicator of not having made a mistake. (Matches JSHint behavior.)
